### DISCUSSION TITLE
Then next big release.  What is it and is it the last for 32 bit machines

### DISCUSSION THREAD

**Discussion by rdstern** [IDEMSInternational/R-Instat#8838] [General]
R-Instat has advanced a lot in recent weeks.  I am keen for it to be reasonably "complete" in the prepare menu, and in describe as far as tables.  
So recently 
a) @lloyddewit has added the main touches to thew scripts that can run.  And that is a wonderful addition.
b) @Patowhiz has made the key breakthrough I hope, on the way to having a system for tables that parallels the great graphics

There are three further elements, but I hope smaller:
c) We need to be able to handle multiple missing values
d) We need to have started on looping
e) I hope improved ODK will also be included?  @EVANSTATS is there progress?

And there continues to be work on improving climatic, improving the graphics, improving the output window and generally making it more solid.

I hope within a month - at most two, we can have a major new release - and one that we can use to propose a lot of funded work for IDEMS and INNODEMS and GHIDEMS?

I note that I would very much like each of the R-Instat development team to also look for somewhere they can present a paper at a local conference.  I'd like to start discussions on this topic in the next week or two. 

The reason for this discussion topic here is that I was assuming this version would also mark the end of our support for 32 bit computers.  We then freeze for 32 bit (called perhaps Version 0.8, etc).  That's because R has advanced and the latest versions of R are only for 64 bit machines.  We then update R, and call it Version 0.9 and continue to develop, but just for 64 bit machines.

We are (I think) still using R 4.1, which is the latest version for 32 bit machines.  The latest version of R is now 4.3.2.

We are likely to be the only R GUI to worry about 32 bit machines.  That's because we remain keen to have R-Instat available everywhere, and we assume that (in some countries) there may continue to be a demand on 32 bit machines.  Another feature that sets R-Instat apart - probably for a similar audience - is that we don't assume good internet access.  (Everyone seems to have some interent access these days, but it is still expensive and therefore many who do not have it all the time.)

My understanding from @ChrisMarsh82 is that it would not be too much trouble to have the frozen version (0.8+) for 32 bit machines and the full development version (0.9 and continuing) for 64 bit machines.  

But Danny has recently questioned the wisdom of this approach.  He argues that the only reason we are moving on, is to keep up with R, but there may (at this stage be almost nothing in the new versions of R that we need?  @lilyclements I wonder if this is true?

If he is right, we don't really need to move on with R and then we keep our lives much simpler by having a single release to support.  He worries that the fixing of bugs, perhaps the updating of packages, etc, might become considerable work.  And we are not at the cutting edge of R.  So, until a new version of R includes features we really need, we should continue with Version 4.1.

**Comment by lilyclements** [comment]
@rdstern I definitely think that @dannyparsons has a much better knowledge and opinion on this than I do! Looking (briefly) at the differences between versions 4.1 and 4.3, I can't see anything that appears to be a major advancement and is therefore a "deal breaker" at this stage.

**Comment by rdstern** [comment]
@lilyclements thanks for that.  I still think we are on a path to a really nice advance with a new release within a couple of months.  Before discussions with Danny I was happy that this could mark the time when we froze a version for 32 bit machines.  Then we can keep up with the new versions of R.  I have switched completely to being much more comfortable that a priority for R-Instat is to cater as well as possible for as many users as we can.  And in some countries they will still have 32 bit machines.  So we keep to a single version for as long as possible.  With David going to Mali and you in Rwanda soon, it could be good to ask whether this issue affects any students, etc. 
So we keep monitoring, and only switch when either there's really exciting additions in R and/or vital packages that we use won't run.
